### PR TITLE
Fix Toml parser test cases in windows

### LINKED ISSUE #20044: Ballerina TOML parser test cases are failing in Windows
**Description:**
The following test cases are failing in Windows. 

Failing with error  `java.lang.NumberFormatException: For input string: "sers"`
- org.ballerinalang.toml.ManifestProcessorTest > testMultipleDependencies
- org.ballerinalang.toml.ManifestProcessorTest > testSingleDependencies

These two are failing because generated TOML contains  `\` (backslash) (from windows path) and those have to be escaped according to the TOML spec. 

 Failing with error ` expected [path/to/github.balo] but found [path\to\github.balo]`
-  ballerina-toml-parser-test-suite > org.ballerinalang.toml.TomlFileToManifestTest > testTomlFile

### PR DESCRIPTION
**Pull request by parkavi11**
Fixes #20044, #20933

## Suggested labels
Area/BuildTools

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 1]
```diff
@@ -131,9 +131,20 @@ public void testSingleDependencies() throws TomlException, IOException {
...
         Path baloPath = tmpDir.resolve("string_utils.balo");
         Files.createFile(baloPath);
-        
-        Manifest manifest = ManifestProcessor.parseTomlContentFromString(this.validProjectBlock + "[dependencies] \n " +
-                "string-utils = {path = \"" + baloPath + "\", version = \"1.1.5\"} \n");
+
+        String baloPathStr = baloPath.toString();
+        String tomlData;
+
+        if (baloPathStr.contains("/")) {
```
Comment (hevayo) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: Shall we use Java Path object to construct the path.

Comment (parkavi11) [reply compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: The path gets modified in windows when Java Path object is used. That's why I used directly the string after replacing with forward slash.

**Comment by pubudu91** [comment]
@hevayo @hemikak Do we need these test cases, since we're using a library for TOML now? If we need these tests, shouldn't it be using the TOML library, instead of using the old, custom written parser?

**Comment by hevayo** [comment]
@pubudu91  ManifestProcessor internally use the Toml library to parse the config. And we can consider theses tests as unit tests for ManifestProcessor. But you are write we can reconsider the test cases. BTW we did find an issue because of these tests.

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 2]
```diff
@@ -131,9 +131,20 @@ public void testSingleDependencies() throws TomlException, IOException {
...
+            tomlData = this.validProjectBlock + "[dependencies] \n " +
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: We can simplify this logic. This line is redundant.

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 3]
```diff
@@ -154,10 +165,23 @@ public void testMultipleDependencies() throws TomlException, IOException {
...
-        
-        Manifest manifest = ManifestProcessor.parseTomlContentFromString(this.validProjectBlock + "[dependencies] \n " +
-                "string-utils = { path = \"" + baloPath + "\", version = \"1.0.5\" } \n " +
-                "jquery = { version = \"2.2.3\" } \n");
+
+        String baloPathStr = baloPath.toString();
+        String tomlData;
+
+        if (baloPathStr.contains("/")) {
+            tomlData = this.validProjectBlock + "[dependencies] \n " +
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: Same here, We can simplify this logic. This line is redundant in both cases.

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java (hunk 2 above)
Comment (parkavi11) [reply compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: simplified

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java (hunk 3 above)
Comment (parkavi11) [reply compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: done

**Pull request closed without merging**